tendermint/src/block.rs

```diff
@@ -84,14 +85,27 @@ impl TryFrom<RawBlock> for Block {
     }
 }
 
-impl From<Block> for RawBlock {
-    fn from(value: Block) -> Self {
-        RawBlock {
-            header: Some(value.header.into()),
+impl TryFrom<Block> for RawBlock {
+    type Error = Error;
+
+    fn try_from(value: Block) -> Result<Self, Error> {
+        Ok(RawBlock {
+            header: Some(value.header.try_into()?),
             data: Some(value.data.into()),
-            evidence: Some(value.evidence.into()),
-            last_commit: value.last_commit.map(Into::into),
-        }
+            evidence: Some(value.evidence.try_into()?),
+            last_commit: value.last_commit.map(|v| v.try_into()).transpose()?,
+        })
+    }
+}
+
+impl Serialize for Block {
```

Comment:
By changing from `TryFrom` to `Try`, it is not possible to auto derive the `Serialize` trait through type attributes. (https://github.com/serde-rs/serde/issues/1524) So I just implement the trait manually.

I'm not sure why there is a need to auto derive the `Serialize` trait other than for use in testing. In any case, I see that there are many custom `impl` blocks for `Serialize` in `tendermint`, so it should be fine to add a few more.